pkg/controller/registry/resolver/resolver_test.go

```diff
@@ -1742,3 +1743,166 @@ func TestInferProperties(t *testing.T) {
 		})
 	}
 }
+
+func TestSortChannel(t *testing.T) {
+	for _, tc := range []struct {
+		Name string
+		In   []*Operator
+		Out  []*Operator
+		Err  error
+	}{
+		{
+			Name: "wrinkle-free",
+			In: []*Operator{
+				{
+					name: "b",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+				{
+					name:     "a",
+					replaces: "b",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+			},
+			Out: []*Operator{
+				{
+					name:     "a",
+					replaces: "b",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+				{
+					name: "b",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+			},
+		},
+		{
+			Name: "empty",
+			In:   nil,
+			Out:  nil,
+		},
+		{
+			Name: "replacement cycle",
+			In: []*Operator{
+				{
+					name:     "a",
+					replaces: "b",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+				{
+					name:     "b",
+					replaces: "a",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+			},
+			Err: errors.New(`no channel heads (entries not replaced by another entry) found in channel "channel" of package "package"`),
+		},
+		{
+			Name: "replacement cycle",
+			In: []*Operator{
+				{
+					name:     "a",
+					replaces: "b",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+				{
+					name:     "b",
+					replaces: "c",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+				{
+					name:     "c",
+					replaces: "b",
+					bundle: &api.Bundle{
+						PackageName: "package",
+						ChannelName: "channel",
+					},
+				},
+			},
+			Err: errors.New(`a cycle exists in the chain of replacement beginning with "a" in channel "channel" of package "package"`),
+		},
+		{
+			Name: "skipped entry omitted",
```

Comment:
This test represents my understanding of the intended behavior of skips here, but it's failing. Is this a bug in the sort or am I mistaken about how this should work?

Reply:
Fixed this and added a higher-level test case (TestSolveOperatorsWithSkipsPreventingSelection) to cover the skips omission behavior.